src/excmds.ts

```diff
@@ -1089,6 +1103,37 @@ document.addEventListener("scroll", addJump, { passive: true })
 //#content_helper
 document.addEventListener("load", () => curJumps().then(() => jumpprev(0)))
 
+// Adds a new entry to history tree or updates it if already visited
+/** @hidden */
+//#content_helper
+export async function addTabHistory() {
+    let pages = await curTabHistory()
+    if (!pages)
+        pages = {
+            current: null,
+            list: [],
+        }
+    const link = getJumpPageId()
+    const current = pages["list"].findIndex(item => item.href === link)
+    if (pages["list"][current]) {
+        pages["current"] = current
+    } else {
+        pages["list"].push({
+            parent: pages["current"],
+            href: link,
+            title: document.title,
+            id: pages["list"].length,
```

Comment:
@GHolk I think we can just use store the system timestamp here, I don't think it should be too complicated